customcommands/interval_test.go

```diff
@@ -1,315 +1,335 @@
+			Cron:    "* * * * 0",
+			Exclude: exclude{Days: []time.Weekday{time.Sunday}},
+			RefTime: now,
+			Want:    neverRuns()},
+
+		{Name: "on minute 5 with exclude hours",
+			Cron:    "5 * * * *",
+			Exclude: exclude{Hours: []int64{0, 1}},
+			RefTime: zero.Add(15 * time.Minute),
+			Want:    runOnDate(hour(2), minute(5))},
+
+		{Name: "on minute 0 on 2nd day with exclude hours",
+			Cron:    "0 * 2 * *",
+			Exclude: exclude{Hours: []int64{0, 1}},
+			RefTime: zero.Add(18 * time.Minute),
+			Want:    runOnDate(day(2), hour(2), minute(0))},
+
+		{Name: "on minute 0 in february with exclude hours",
+			Cron:    "0 * * 2 *",
+			Exclude: exclude{Hours: []int64{0, 1}},
+			RefTime: zero,
+			Want:    runOnDate(month(time.February), day(1), hour(2), minute(0))},
+
+		{Name: "on minute 0 on tues with exclude hours",
+			Cron:    "0 * * * 2",
+			Exclude: exclude{Hours: []int64{0, 1}},
+			RefTime: zero,
+			Want:    runOnDate(day(2), hour(2), minute(0))},
+
+		{Name: "on minute 5 with exclude days",
+			Cron:    "5 * * * *",
+			Exclude: exclude{Days: []time.Weekday{time.Sunday, time.Monday}},
+			RefTime: zero,
+			Want:    runOnDate(day(2), hour(0), minute(5))},
+
+		{Name: "on minute 0 on day 2,3 with exclude days",
+			Cron:    "0 * 2,3 * *",
+			Exclude: exclude{Days: []time.Weekday{time.Sunday, time.Monday}},
+			RefTime: zero,
+			Want:    runOnDate(day(2), hour(0), minute(0))},
+
+		{Name: "on minute 0 of day 1 in feb,mar,apr with exclude days",
+			Cron:    "0 * 1 2,3,4 *",
+			Exclude: exclude{Days: []time.Weekday{time.Thursday}},
+			RefTime: zero,
+			Want:    runOnDate(month(time.April), day(1), hour(0), minute(0))},
+
+		{Name: "on minute 0 on tues,wed with exclude days",
+			Cron:    "0 * * * 2,3",
+			Exclude: exclude{Days: []time.Weekday{time.Tuesday}},
+			RefTime: zero,
+			Want:    runOnDate(day(3), hour(0), minute(0))},
+
+		{Name: "on minute 5 with exclude days and hours",
+			Cron:    "5 * * * *",
+			Exclude: exclude{Days: []time.Weekday{time.Tuesday}, Hours: allHoursExcept(1)},
+			RefTime: zero.Add(2 * time.Hour),
+			Want:    runOnDate(day(3), hour(1), minute(5))},
+
+		{Name: "on minute 0 of day 1,2,3 with exclude days and hours",
+			Cron:    "0 * 1,2,3 * *",
+			Exclude: exclude{Days: []time.Weekday{time.Tuesday}, Hours: allHoursExcept(1)},
+			RefTime: zero.Add(2 * time.Hour),
+			Want:    runOnDate(day(3), hour(1), minute(0))},
+
+		{Name: "on minute 0 of first day of jan,feb,mar,apr with exclude days and hours",
+			Cron:    "0 * 1 1,2,3,4 *",
+			Exclude: exclude{Days: []time.Weekday{time.Thursday}, Hours: allHoursExcept(1)},
+			RefTime: zero.Add(2 * time.Hour),
+			Want:    runOnDate(month(time.April), day(1), hour(1), minute(0))},
+
+		{Name: "on minute 0 of mon,tue,wed with exclude days and hours",
+			Cron:    "0 * * * 1,2,3",
+			Exclude: exclude{Days: []time.Weekday{time.Tuesday}, Hours: allHoursExcept(1)},
+			RefTime: zero.Add(2 * time.Hour),
+			Want:    runOnDate(day(2), hour(1), minute(0))},
```

Comment:
Shouldn’t this be expected to run on Day 3?

Reply:
Whoops! I changed this temporarily to demonstrate the test output on failures, but forgot to change things back.